activerecord/lib/active_record/model_schema.rb

```diff
@@ -484,6 +484,9 @@ def load_schema
         end
 
         def load_schema!
+          unless table_exists?
+            raise ActiveRecord::NoTableError, "columns_hash is not available for a model with no table"
```

Comment:
This error still mentions `columns_hash`, but `load_schema!` is called from several other places.

```suggestion
            raise ActiveRecord::NoTableError, "#{self} has no table configured"
```

Reply:
❤️